docker/wholegraph/Dockerfile

```diff
@@ -0,0 +1,113 @@
+# Docker file for building a docker image for running GraphStorm code on Amazon SageMaker
+# Note: Distributed graph partition will use another docker image which will come soon.
+
+ARG DEVICE=gpu
+
+FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.0.1-gpu-py310-cu118-ubuntu20.04-ec2
+ENV dev_type=GPU
+# Install DGL GPU version
+RUN pip3 install dgl==1.0.4+cu117 -f https://data.dgl.ai/wheels/cu117/repo.html && rm -rf /root/.cache
+
+LABEL maintainer="Amazon AI Graph ML team"
+
+# Install related Python packages
+RUN pip3 install ogb==1.3.6 scipy pyarrow boto3 scikit-learn transformers
+
+# Install other dependencies
+RUN apt-get update
+RUN apt-get install -y apt-transport-https
+RUN apt-get install -y cython3 libicu-dev
+RUN pip3 install h5py psutil
+
+RUN apt-get install -y unzip
+
+# Install GraphStorm from source code
+RUN mkdir -p /graphstorm
+COPY code/python/graphstorm /graphstorm/python/graphstorm
+ENV PYTHONPATH="/graphstorm/python/:${PYTHONPATH}"
+
+# Copy GraphStorm scripts and tools
+COPY code/inference_scripts /graphstorm/inference_scripts
+COPY code/tools /graphstorm/tools
+COPY code/training_scripts /graphstorm/training_scripts
+
+RUN ls /usr/local/cuda*
+
+#################################################
+## Install NCCL
+RUN apt-get clean
+RUN apt-get reinstall ca-certificates
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/libnccl2_2.15.1-1+cuda11.8_amd64.deb
+RUN dpkg -i libnccl2_2.15.1-1+cuda11.8_amd64.deb
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/libnccl-dev_2.15.1-1+cuda11.8_amd64.deb
+RUN dpkg -i libnccl-dev_2.15.1-1+cuda11.8_amd64.deb
```

Comment:
Let's follow the step 6 from the [EC2 doc ](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start-nccl-base.html#nccl-start-base-nccl)to install NCCL.